### PR TITLE
[15017] Adding empty implementation for ExternalLocatorsProcessor

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR adds an empty implementation for `ExternalLocatorsProcessor`, so we can add unit tests for it and, in parallel, prepare the implementation

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by richiware — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/network/ExternalLocatorsProcessor.hpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+namespace ExternalLocatorsProcessor {
+
+using eprosima::fastrtps::rtps::LocatorSelectorEntry;
+using eprosima::fastrtps::rtps::ParticipantProxyData;
+using eprosima::fastrtps::rtps::ReaderProxyData;
+using eprosima::fastrtps::rtps::WriterProxyData;
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.hpp:37]: `ExternalLocatorsProcessor` is a namespace, not an object.

File: src/cpp/rtps/network/ExternalLocatorsProcessor.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ */
+void add_external_locators(
+        ReaderProxyData& data,
+        const ExternalLocators& external_locators);
+
+/**
+ * Filter the locators of a remote participant according to the matching algorithm.
+ *
+ * @param [in,out]  data                           ParticipantProxyData of the remote participant to update.
+ * @param [in]      metatraffic_external_locators  The external locators collection to use for filtering of the
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.hpp:87]: I think passive does not apply here.

File: src/cpp/rtps/network/ExternalLocatorsProcessor.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ *                                                 meta-traffic locators.
+ * @param [in]      default_external_locators      The external locators collection to use for filtering of the
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.hpp:89]: I think passive does not apply here.

File: src/cpp/rtps/network/ExternalLocatorsProcessor.hpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        ParticipantProxyData& data,
+        const ExternalLocators& metatraffic_external_locators,
+        const ExternalLocators& default_external_locators,
+        bool ignore_non_matching);
+
+/**
+ * Filter the locators of a remote endpoint according to the matching algorithm.
+ *
+ * @param [in,out]  locators             LocatorSelectorEntry of the remote destination to update.
+ * @param [in]      external_locators    The external locators collection to use for filtering of the locators.
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/network/ExternalLocatorsProcessor.hpp:104]: I think passive does not apply here.

**Review by richiware — APPROVED** [review]

**Pull request merged**